Venmo/src/main/java/com/braintreepayments/api/VenmoClient.java

```diff
@@ -428,6 +445,196 @@ private Intent getLaunchIntent(Configuration configuration, String profileId, St
         return venmoIntent;
     }
 
+    /**
+     * Use this method with the manual browser switch integration pattern.
+     *
+     * @param browserSwitchResult a {@link BrowserSwitchResult} with a {@link BrowserSwitchStatus}
+     * @param callback            {@link VenmoOnActivityResultCallback}
+     */
+    public void onBrowserSwitchResult(@NonNull BrowserSwitchResult browserSwitchResult, @NonNull final VenmoOnActivityResultCallback callback) {
+        int result = browserSwitchResult.getStatus();
+        switch (result) {
+            case BrowserSwitchStatus.CANCELED:
+                braintreeClient.sendAnalyticsEvent("pay-with-venmo.browser-switch.canceled");
+                callback.onResult(null, new UserCanceledException("User canceled Venmo."));
+                break;
+            case BrowserSwitchStatus.SUCCESS:
+                Uri deepLinkUri = browserSwitchResult.getDeepLinkUrl();
+                if (deepLinkUri != null) {
+                    if (deepLinkUri.getPath().contains("success")) {
+                        braintreeClient.sendAnalyticsEvent("pay-with-venmo.browser-switch.success");
+                        String resourceIdFromBrowserSwitch = Uri.parse(String.valueOf(deepLinkUri)).getQueryParameter("resource_id");
+                        String paymentMethodNonceFromBrowserSwitch = Uri.parse(String.valueOf(deepLinkUri)).getQueryParameter("payment_method_nonce");
+                        String usernameFromBrowserSwitch = Uri.parse(String.valueOf(deepLinkUri)).getQueryParameter("username");
+
+                        String resourceIdFromAppSwitch = "";
```

Comment:
```suggestion
                        String resourceIdFromAppSwitch;
```

Do we need these to be defined as empty strings? It seems like if not set they should be null

Reply:
It seems like the compiler doesn't like that syntax, but does like `String resourceIdFromAppSwitch = null;` for defaulting to null?

Reply:
Updated here: https://github.com/braintree/braintree_android/pull/911/commits/a25792e1bf8098124108d1d0c90455b1e94b71f8